### DISCUSSION TITLE
Is it possible to use Differential Privacy on the DoppelGANger architecture?

### DISCUSSION THREAD

**Discussion by geheim01** [gretelai/gretel-synthetics#152] [Q&A] [answered]
I want to train a DoppelGANger architecture and apply Differential Privacy to synthesize privacy-preserving time series data.
I don't see the option to apply DP in the DGANConfig class. Am I missing something or do I need to fork the project and change the optimizer to the TensorFlow DPOptimizer?

**Comment by lipikaramaswamy** [comment, marked as answer]
Hello! We do not offer differential privacy with DoppelGANger at present. You could fork the repo and swap out the optimizer in the discriminator with a differentially private one from [Opacus](https://opacus.ai/), the DP optimizer library for torch. Note that the generator's optimizer needn't be changed since it never directly uses the training data, and would be covered by the post processing property of DP on the discriminator. Let us know how it goes!
